server/middleware/authorization.js

```diff
@@ -0,0 +1,158 @@
+import auth from '../helpers/auth';
+import response from '../helpers/response';
+
+class authorization {
+  /**
+   * Helps create user type
+   *
+   * @static createUserType
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @memberof authorization
+   */
+  static async createUserType(req, res, next) {
+    // Checks for token in the header
+    const headerAuth = req.header('Authorization');
+
+    // Sets User type to client if no token found
+    if (!headerAuth && !req.body.token) {
+      req.body.type = 'client';
+      req.isAdmin = null;
+      next();
+      return;
+    }
+
+    // Extracts token when given
+    const token = req.body.token || headerAuth.split(' ')[1];
+
+    // sets placehokder for successfully decoded token
+    let decoded;
+
+    try {
+      decoded = await auth.decodeToken(token);
+    } catch (err) {
+      // if Error returns authentication failed so user can re authenticate
+      return response.error(res, 401, 'Token Invalid');
+    }
+
+    const { type } = decoded;
+    const { isAdmin } = req.body;
+
+    req.body.type = type === 'admin' ? 'staff' : 'client';
+    req.body.isAdmin = type === 'admin' ? isAdmin : null;
+
+    next();
+  }
+
+  /**
+   * Checks if token passed are valid
+   *
+   * @static requireToken
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns
+   * @memberof authorization
+   */
+  static async requireToken(req, res, next) {
```

Comment:
@swisskid95 please naming conventions. `req` and `res` are not the best of names. Please change this. Request and response are better names.

Reply:
Okay, I just did the needful